Metalama.Framework/Advising/AdviserExtensions.cs

```diff
@@ -0,0 +1,1104 @@
+    ///     if the getter should not be overridden. This method must be annotated with <see cref="TemplateAttribute"/>. The signature of this method must
+    ///     be <c>T Get()</c> where <c>T</c> is either <c>dynamic</c> or a type compatible with the type of the field or property.
+    ///     To select a different templates for iterator getters, use the constructor of the <see cref="GetterTemplateSelector"/> type. To specify a single
+    ///     template for all properties, pass a string.
+    /// </param>
+    /// <param name="setTemplate">The name of the method of the aspect class whose implementation will be used as a template for the getter, or <c>null</c>
+    ///     if the getter should not be overridden. This method must be annotated with <see cref="TemplateAttribute"/>. The signature of this method must
+    ///     be <c>void Set(T value</c>  where <c>T</c> is either <c>dynamic</c> or a type compatible with the type of the field or property.</param>
+    /// <param name="args">An object (typically of anonymous type) whose properties map to parameters or type parameters of the template methods.</param>
+    /// <param name="tags">An optional opaque object of anonymous type passed to the template method and exposed under the <see cref="meta.Tags"/> property of the
+    ///     <see cref="meta"/> API.</param>
+    /// <seealso href="@overriding-fields-or-properties"/>
+    public static IOverrideAdviceResult<IPropertyOrIndexer> OverrideAccessors(
+        this IAdviser<IFieldOrPropertyOrIndexer> adviser,
+        in GetterTemplateSelector getTemplate = default,
+        string? setTemplate = null,
+        object? args = null,
+        object? tags = null )
+        => ((IAdviserInternal) adviser).AdviceFactory.OverrideAccessors(
+            adviser.Target,
+            getTemplate,
+            setTemplate,
+            args,
+            tags );
+
+    /// <summary>
+    /// Introduces a field to the target type by specifying a template.
+    /// </summary>
+    /// <param name="adviser">An adviser for a member or a named type.</param>
+    /// <param name="template">Name of the introduced field.</param>
+    /// <param name="scope">Determines the scope (e.g. <see cref="IntroductionScope.Instance"/> or <see cref="IntroductionScope.Static"/>) of the introduced
+    ///     field. The default scope is <see cref="IntroductionScope.Instance"/>.</param>
+    /// <param name="whenExists">Determines the implementation strategy when a property of the same name is already declared in the target type.
+    ///     The default strategy is to fail with a compile-time error.</param>
+    /// <param name="buildField"></param>
+    /// <param name="tags"></param>
+    /// <returns>An <see cref="IPropertyBuilder"/> that allows to dynamically change the name or type of the introduced property.</returns>
+    /// <seealso href="@introducing-members"/>
+    public static IIntroductionAdviceResult<IField> IntroduceField(
```

Comment:
I think it would be useful to have Introduce* overloads without the template and with a mandatory builder. Especially for fields and auto properties.

Reply:
I don't see the reason of making the builder mandatory since all we need is the name and the type, the rest is optional.